viser/client/src/Titlebar.tsx

```diff
@@ -0,0 +1,176 @@
+import Box from "@mui/material/Box";
+import { Button, Grid, Icon, IconButton, SvgIcon } from "@mui/material";
+import { useContext } from "react";
+import { ViewerContext } from ".";
+
+import * as Icons from '@mui/icons-material';
+type IconName = keyof typeof Icons;
+
+const buttonVariants = ['text', 'contained', 'outlined'];
+function isVariant(variant: unknown): variant is ('text' | 'contained' | 'outlined') {
+    return typeof variant === 'string' && buttonVariants.includes(variant);
+}
+
+interface TitlebarButtonProps {
+    text?: string;
+    href?: string;
+    icon?: string;
+    variant?: string;
+}
+function isButtonData(obj: any): obj is TitlebarButtonProps {
```

Comment:
if we strengthen the types on the theme message, it seems like we could do away with these type guards entirely? maybe we could lean on type narrowing via a switch/case like how `WebsocketInterface.tsx` handles all of the possible incoming message types

with the current setup + type guards there's quite a bit of repetition + room for possible errors that `tsc` won't catch